handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+!!!note "Storing Captured Data"
+    For upgrades and migrations for a SystemLink Server with a small amount of data you may be able to use `nislmigrate` to backup your data to the local file system of your SystemLink Server. For SystemLink Servers with a large amount of data be aware `nislmigrate` could exhaust the local storage on the server. This can mitigated by using an attached volume. **This document assumes this will be needed and refers to this attached volume as `D:\` in each workflow.**
+
+## Single Node Upgrade
+
+This workflow describes steps to upgrade a single node deployment of SystemLink Server.
+
+The SystemLink NI Package Manager Installers (NIPM) supports in-place upgrades of SystemLink Server. An in place upgrade is one where the upgrade is run directly on your current SystemLink Server. For production deployments of SystemLink Server this is not recommended. If you pursue this option it is highly recommended you backup your server prior to the upgrade. This backup is an essential fallback should an issue occur during upgrade that leaves your SystemLink Server in a bad state.
+
+For single node upgrades, NI recommends pairing the upgrade with a migration. This mitigates risks of issues during the upgrade by ensuring your original SystemLink Server remains in an operable state.
+
+1. Backup your SystemLink Server.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --all --secret <your secret> --dir D:\migration`.
+
+    !!!note
+        While `nislmigrate` runs all SystemLink services are stopped. Plan ahead for this downtime of your SystemLink server. When the capture is completed, SystemLink services will be restarted automatically.
+
+1. Detach `D:\`.
+
+    !!!note
+        At this point in the workflow your original SystemLink Server is in an operable state. Be aware if any new data is created or ingested by SystemLink Server at this time it will not be available to your new server.
+
+1. Provision a new Window server for SystemLink.
+
+1. Install and configure the new version of SystemLink Server.
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink Server.
+
+1. Run the command `nislmigrate restore --all --secret <your secret> --dir D:\migration`.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
+
+## Single Node to Multi Node Migration
+
+This section describes workflows used when you intend to upgrade or migrate from a single node SystemLink Server configuration to a multi node SystemLink Server configuration that makes use of dedicated servers for MongoDB, PostgreSQL, or file storage. While not required, these migration workflows include provisioning a new SystemLink Server. This is done to reduce risk and ensure your original SystemLink Server is always in an operable state.
```

Comment:
I would just mark the step about provisioning as optional rather than calling it out here and having the user jump around mentally before they start the process.

Suggested rewrite:

This section describes workflows to prepare to upgrade or migrate from a single node SystemLink Server configuration to a multi node SystemLink Server configuration that makes use of dedicated servers for MongoDB, PostgreSQL, or file storage.

Reply:
I like the suggestions. Included your rewrite and put the comment about options in the context of the workflow step.